mm2src/coins/z_coin.rs

```diff
@@ -488,11 +485,14 @@ impl ZCoin {
             tx_builder.add_tx_out(output);
         }
 
+        #[cfg(not(target_arch = "wasm32"))]
         let (tx, _) = async_blocking({
             let prover = self.z_fields.z_tx_prover.clone();
             move || tx_builder.build(consensus::BranchId::Sapling, prover.as_ref())
         })
         .await?;
+        #[cfg(target_arch = "wasm32")]
+        let (tx, _) = tx_builder.build(consensus::BranchId::Sapling, self.z_fields.z_tx_prover.clone().as_ref())?;
```

Comment:
I haven't tested this function in `WASM` yet...for the comment #2000, I notice the issue while testing <https://github.com/KomodoPlatform/komodo-defi-framework/blob/f530d757714b931c942241eda013c03273d20b08/mm2src/coins/z_coin/storage/walletdb/wasm/mod.rs#L725> which uses `create_spend_to_address` directly from `zcash` lib

Reply:
I see, we will likely need Web Workers for this operation, as it may take significant time even on native platforms.

Reply:
How about using `spawn_local_abortable` and channels for now (in this PR) until we use workers for multi-threading https://github.com/KomodoPlatform/komodo-defi-framework/blob/a05819d272278fc708a5ee42a6004edc80ec0e80/mm2src/common/executor/wasm_executor.rs#L34
For inspirations
https://github.com/KomodoPlatform/komodo-defi-framework/blob/4a93790e01a1227d97cdb774fa52f34a694785cf/mm2src/hw_common/src/transport/webusb_driver.rs#L125-L145

Reply:
alright thanks, I'll look into this this weekend or next week

Reply:
done @shamardy

Reply:
can you please use `spawn_local_abortable` instead of `spawn_local`, ref. https://github.com/KomodoPlatform/komodo-defi-framework/blob/a05819d272278fc708a5ee42a6004edc80ec0e80/mm2src/common/executor/wasm_executor.rs#L22-L25